[Swift in WebKit] Work towards modularizing WebCore private headers (Part 3)

rr-codes opened the pull request:
#### d4e2d9ea336ddfd97929073f520c20f8c77a6c92
<pre>
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 3)
<a href="https://bugs.webkit.org/show_bug.cgi?id=297696">https://bugs.webkit.org/show_bug.cgi?id=297696</a>
<a href="https://rdar.apple.com/158807128">rdar://158807128</a>

Reviewed by Abrar Rahman Protyasha.

Add some missing `wtf/Platform.h` includes. This is required because currently, any WebCore client
must themselves manually import `wtf/Platform.h` whenever they import a WebCore header. The module
verifier correctly catches this violation, and so each private header that needs one of the macros
also needs to import `wtf/Platform.h` file.

* Source/WebCore/

Add missing wtf/Platform.h includes, via a generated script.

Canonical link: <a href="https://commits.webkit.org/299025@main">https://commits.webkit.org/299025@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/501727d36e3e260eb34ab6161462db84e77ea338

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117423 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37093 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27712 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123522 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69438 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1eab185c-3498-47b0-87c2-468e7841e336) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119301 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37789 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45680 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89108 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43788 "Too many flaky failures: css3/filters/drop-shadow-current-color.html, fast/canvas/offscreen-no-script-context-crash.html, http/tests/permissions/storage-access-permissions-query.html, http/tests/resourceLoadStatistics/only-accept-first-party-cookies-with-third-party-cookie-blocking.html, imported/w3c/web-platform-tests/event-timing/contextmenu.html, imported/w3c/web-platform-tests/trusted-types/TrustedTypePolicyFactory-getAttributeType-event-handler-content-attributes.tentative.html, imported/w3c/web-platform-tests/trusted-types/set-event-handlers-content-attributes.tentative.html, storage/indexeddb/modern/index-4-private.html, storage/indexeddb/objectstore-cursor.html, webgl/2.0.y/conformance/ogles/GL/mat/mat_041_to_046.html, webgl/2.0.y/conformance/ogles/GL/notEqual/notEqual_001_to_008.html, webgl/2.0.y/conformance2/textures/webgl_canvas/tex-2d-rgb9_e5-rgb-half_float.html, workers/btoa-oom.html, workers/sab/growable-shared-array-buffer-serialization.html (failure)") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/144390d6-24ee-4282-92d6-1cbe812732d0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120375 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30079 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105282 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69617 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/13fb67bf-9aba-4790-99c7-86ce94dff935) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29135 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23399 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67195 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99482 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23581 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126643 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44320 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33312 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97772 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44678 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101519 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97566 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24858 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42936 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20872 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/40670 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44193 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49852 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43650 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46994 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45345 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b902d1cc3c0636372acdb919ae34c96513b69fa6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49686

rr-codes committed:
[Swift in WebKit] Work towards modularizing WebCore private headers (Part 3)
https://bugs.webkit.org/show_bug.cgi?id=297696
rdar://158807128

Reviewed by Abrar Rahman Protyasha.

Add some missing `wtf/Platform.h` includes. This is required because currently, any WebCore client
must themselves manually import `wtf/Platform.h` whenever they import a WebCore header. The module
verifier correctly catches this violation, and so each private header that needs one of the macros
also needs to import `wtf/Platform.h` file.

* Source/WebCore/

Add missing wtf/Platform.h includes, via a generated script.

Canonical link: https://commits.webkit.org/299025@main

weinig commented:
The idiom up until now was that only WebKit could include WebCore headers and all WebKit implementation files already include Platform.h through config.h. 

I worry this will break in the future unless all the ports can enforce this at compile time. 

At the very least, Platform.h should be the first include, as all other files require it.
